openvpn/client/cliproto.hpp

```diff
@@ -443,12 +443,27 @@ class Session : ProtoContextCallbackInterface,
             if (buf.size())
             {
                 const ProtoContext::ProtoConfig &c = proto_context.conf();
+
+                const uint8_t* packet_data = buf.data();
```

Comment:
Looking at [ptb.hpp](https://github.com/OpenVPN/openvpn3/blob/master/openvpn/ip/ptb.hpp#L45), you could use indexing operator and do `buf[0]`.